src/os/bluestore/bluestore_types.h

```diff
@@ -458,21 +472,48 @@ struct bluestore_blob_t {
   bluestore_blob_t(uint32_t f = 0) : flags(f) {}
 
   DENC_HELPERS;
+
+  BLUESTORE_FORCEINLINE
   void bound_encode(size_t& p, uint64_t struct_v) const {
-    assert(struct_v == 1 || struct_v == 2);
-    denc(extents, p);
-    denc_varint(flags, p);
-    denc_varint_lowz(compressed_length_orig, p);
-    denc_varint_lowz(compressed_length, p);
-    denc(csum_type, p);
-    denc(csum_chunk_order, p);
-    denc_varint(csum_data.length(), p);
-    p += csum_data.length();
-    p += sizeof(unsigned long long);
+    size_t p_ct = 0;
```

Comment:
TODO: many different accumulators _might_ help minimise dependencies between instructions, and thus leverage OOO facilities in CPU. However, it can be that compiler already does that for us. Need to check.